### PR TITLE
#2445 tests

### PR DESCRIPTION
**Pull request by stenington**
Trying to get rid of deprecation warnings by upgrading capybara.

### CODE REVIEW

**Comment by stenington** [comment]
@shaunxp20 I probably should have had you review my initial changes without adding any new tests, so let me know if you want me to roll back my latest commits. But this PR updates capybara to get rid of the deprecation warnings, fixes a few tests that broke as a result of that upgrade, and introduces a new system spec for students submitting.

File: spec/system/student/submit_a_project_spec.rb [hunk 1]
```diff
@@ -0,0 +1,94 @@
+require "rails_helper"
```
Comment (shaunxp20) [inline spec/system/student/submit_a_project_spec.rb]: Regarding our convo about what goes in `feature/` vs `system/`. Just making this up as I go, but I wonder if this style of test goes in `feature/`? Anywho, just a suggestion to help draw that line. #no-big-deal #random-thought 😃

File: spec/system/student/submit_a_project_spec.rb [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    given_there_is_a_team_with_a_complete_submission
+    that_hasnt_been_submitted
+    and_a_student_from_that_team_is_logged_in
+
+    when_they_review_it_from_the_dashboard
+    and_submit_it
+
+    then_they_see_that_their_app_will_be_judged
+    and_it_is_marked_as_published
+  end
```
Comment (shaunxp20) [inline spec/system/student/submit_a_project_spec.rb]: Woah! I really like the way these specs read!!!! :100: 😍 🙌 🏅 💯

File: spec/system/student/submit_a_project_spec.rb (hunk 1 above)
Comment (stenington) [reply spec/system/student/submit_a_project_spec.rb]: I found [this post](https://rossta.net/blog/why-rails-system-tests-matter.html) discussing it. It looks like system tests are a new concept with rails 5.1, and they allow capybara tests to run in transactions which previously wasn't possible. So it seems to me like the major difference is a technical aspect of database cleanup.

I'm thinking of moving this test to `feature/` and defaulting to putting stuff there.

**Pull request merged**